mmv1/products/firebasehosting/CustomDomain.yaml

```diff
@@ -0,0 +1,514 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+!ruby/object:Api::Resource
+name: "CustomDomain"
+min_version: beta
+base_url: projects/{{project}}/sites/{{site_id}}/customDomains
+self_link: projects/{{project}}/sites/{{site_id}}/customDomains/{{custom_domain}}
+create_url: projects/{{project}}/sites/{{site_id}}/customDomains?customDomainId={{custom_domain}}
+update_verb: :PATCH
+update_mask: true
+description: |
+  Manages Custom Domains for Firebase Hosting. Custom Domains link your
+  domain names with Firebase Hosting sites, allowing Hosting to serve content
+  on those domain names.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    "Official Documentation": "https://firebase.google.com/docs/hosting"
+  api: "https://firebase.google.com/docs/reference/hosting/rest/v1beta1/projects.sites.customdomains"
```

Comment:
Note to make sure this URL is reachable once the API is public

Reply:
Done.